core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/clustering/hazelcast/kubernetes/KubernetesMembershipScheme.java

```diff
@@ -0,0 +1,164 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.core.clustering.hazelcast.kubernetes;
+
+import com.hazelcast.cluster.Member;
+import com.hazelcast.cluster.MembershipEvent;
+import com.hazelcast.cluster.MembershipListener;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.NetworkConfig;
+import com.hazelcast.config.TcpIpConfig;
+import com.hazelcast.core.HazelcastInstance;
+import org.apache.axis2.clustering.ClusteringMessage;
+import org.apache.axis2.description.Parameter;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.core.clustering.hazelcast.HazelcastCarbonClusterImpl;
+import org.wso2.carbon.core.clustering.hazelcast.HazelcastConstants;
+import org.wso2.carbon.core.clustering.hazelcast.HazelcastMembershipScheme;
+import org.wso2.carbon.core.clustering.hazelcast.HazelcastUtil;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Registering the configurations required for Hazelcast auto discovery plugin in kubernetes and define membership schemes.
+ */
+public class KubernetesMembershipScheme implements HazelcastMembershipScheme {
+
+    private static final Log log = LogFactory.getLog(KubernetesMembershipScheme.class);
+
+    private final Map<String, Parameter> parameters;
+    private final NetworkConfig nwConfig;
+    private final List<ClusteringMessage> messageBuffer;
+    private HazelcastInstance primaryHazelcastInstance;
+    private HazelcastCarbonClusterImpl carbonCluster;
+
+    public KubernetesMembershipScheme(Map<String, Parameter> parameters, String primaryDomain, Config config,
+                                      HazelcastInstance primaryHazelcastInstance, List<ClusteringMessage> messageBuffer) {
+
+        this.parameters = parameters;
+        this.primaryHazelcastInstance = primaryHazelcastInstance;
+        this.messageBuffer = messageBuffer;
+        this.nwConfig = config.getNetworkConfig();
+    }
+
+    @Override
+    public void setPrimaryHazelcastInstance(HazelcastInstance primaryHazelcastInstance) {
+
+        this.primaryHazelcastInstance = primaryHazelcastInstance;
+    }
+
+    @Override
+    public void setLocalMember(Member localMember) {
+
+    }
+
+    @Override
+    public void setCarbonCluster(HazelcastCarbonClusterImpl hazelcastCarbonCluster) {
+
+        this.carbonCluster = hazelcastCarbonCluster;
+    }
+
+    @Override
+    public void init() {
+
+        String localMemberPort = "4000";
```

Comment:
Let's use a constant for this

Reply:
Fixed by - https://github.com/wso2/carbon-kernel/pull/3429